lib/rules/id-length.js

```diff
@@ -63,8 +70,21 @@ module.exports = {
 
                 return obj;
             }, {});
+        const exceptionPatterns = options.exceptionPatterns || [];
         const reportedNode = new Set();
 
+        /**
+         * Checks if a string matches the provided exception patterns
+         * @param {string} name The string to check.
+         * @returns {boolean} if the string is a match
+         * @private
+         */
+        function matchesExceptions(name) {
+            return exceptionPatterns.some(
+                pattern => new RegExp(pattern, "u").test(name)
+            );
```

Comment:
I think it would be better to create all RegExp instance in the "create(context)", because it's not changed after generated from the option "exceptionPatterns".